cmd/av/helpers.go

```diff
@@ -165,3 +165,47 @@ func branchNameArgs(
 	branches, _ := allBranches(cmd.Context())
 	return branches, cobra.ShellCompDirectiveNoSpace
 }
+
+// validateBranchSync checks if the current branch is in sync with its parent branch.
+// Only enforces this check for non-trunk branches when we have a stored parent head.
+// This validation prevents commands from operating on branches that are out of sync
+// with their parent, which could lead to unexpected behavior or loss of commits.
+func validateBranchSync(ctx context.Context, repo *git.Repo, branch meta.Branch) error {
```

Comment:
Not sure the philosophy around how much this CLI should act as a wizard. I.e. rather than throwing an error maybe the cli should prompt for "do you want to run av sync?". 

Happy to update depending on what you prefer